tests/test_provider.py

```diff
@@ -128,3 +135,118 @@ def test_logger_changes_should_cascade_to_evaluation_converter(provider: LaunchD
 
     assert len(caplog.records) == 1
     assert caplog.records[0].message == "'kind' was set to a non-string value; defaulting to user"
+
+
+def test_provider_emits_ready_event_when_immediately_ready():
+    ld_provider_ready_count = 0
+    lock = threading.Lock()
+
+    def handle_status(details: EventDetails):
+        if details.provider_name == 'launchdarkly-openfeature-server':
+            nonlocal lock
+            nonlocal ld_provider_ready_count
+            with lock:
+                ld_provider_ready_count = ld_provider_ready_count + 1
+
+    # At the time of implementation this handler runs synchronously on the same
+    # thread as initialization. The lock is in case this behavior changes.
+    api.add_handler(ProviderEvent.PROVIDER_READY, handle_status)
+
+    openfeature_provider = LaunchDarklyProvider(Config("", offline=True))
+    api.set_provider(openfeature_provider)
+
+    with lock:
+        assert ld_provider_ready_count == 1
+
+    api.shutdown()
+
+
+def test_provider_emits_error_event_immediately_failed():
+    ld_provider_error_count = 0
+    lock = threading.Lock()
+
+    def handle_status(details: EventDetails):
+        if details.provider_name == 'launchdarkly-openfeature-server':
+            nonlocal lock
+            nonlocal ld_provider_error_count
+            with lock:
+                ld_provider_error_count = ld_provider_error_count + 1
+
+    api.add_handler(ProviderEvent.PROVIDER_ERROR, handle_status)
+
+    openfeature_provider = LaunchDarklyProvider(
+        Config("", update_processor_class=FailingDataSource, send_events=False))
+
+    try:
+        api.set_provider(openfeature_provider)
+        # The method should throw
+        assert False
```

Comment:
I think you can also do this, which is probably slightly preferred. 
```suggestion
        pytest.fail("set_provider should have raised an exception")
```